docs/fundamentals/productivity/code-analysis.md

```diff
@@ -35,11 +40,24 @@ The following rules are enabled, by default, in .NET 5.0 Preview 8.
 | [CA2015](/visualstudio/code-quality/ca2015) | Reliability | Warning | Do not define finalizers for types derived from <xref:System.Buffers.MemoryManager%601> |
 | [CA2247](/visualstudio/code-quality/ca2247) | Usage | Warning | Argument passed to TaskCompletionSource constructor should be <xref:System.Threading.Tasks.TaskCreationOptions> enum instead of <xref:System.Threading.Tasks.TaskContinuationOptions> |
 
-You can change the severity of these rules to disable them or elevate them to errors. In addition, you may want to enable additional rules that are included in this release but are disabled by default. For more information, see [Configure code analysis rules](configure-code-analysis-rules.md).
+You can change the severity of these rules to disable them or elevate them to errors.
+
+### Enabling additional rules
+
+Starting with .NET 5.0 RC2, the .NET SDK ships with [all "CA" code quality rules](/visualstudio/code-quality/code-analysis-for-managed-code-warnings). For a full list of rules that are included with each .NET SDK version, see [analyzer releases](https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md).
+
+#### Default analysis mode
 
-For a full list of rules that are included with each .NET SDK version, see [analyzer releases](https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md). This list also includes each rule's default severity. Some rules are enabled only as IDE suggestions with corresponding code fixes.
+In the default analysis mode, some rules are [enabled by default](#enabled-rules) as build warnings. Some other rules are enabled by default only as Visual Studio IDE suggestions with corresponding code fixes. Rest of the rules are disabled by default. Full list of rules specified at [analyzer releases](https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md) includes each rule's default severity and whether or not the rule is enabled by default in the default analysis mode.
 
-## Treat warnings as errors
+#### Custom analysis mode
+
+You can [configure code analysis rules](configure-code-analysis-rules.md) to enable or disable individual rule or a category of rules. Additionally, you can configure the [AnalysisMode](../../core/project-sdk/msbuild-props.md#analysismode) to switch to below custom analysis modes:
+
+- _Aggressive_ or _Opt-out_ mode: All rules are enabled by default as build warnings. You can selectively [opt out](configure-code-analysis-rules.md) of individual rules to disable them.
+- _Conservative_ or _Opt-in_ mode, where all rules are disabled by default. You can selectively [opt into](configure-code-analysis-rules.md) individual rules to enable them.
+
+### Treat warnings as errors
```

Comment:
If this property only applies to code quality warnings and not code style warnings, I think we should state it explicitly.

Reply:
Good point. I agree with your perception that in general it might be better for this property to apply to both quality and style warnings, though this will be a separate feature request in Roslyn. For now, I'll update the docs to reflect the current semantics.